tests/test_plugin.py

```diff
@@ -140,3 +145,18 @@ def test_account(fxa_urls):
     """)
     result = testdir.runpytest()
     result.assert_outcomes(passed=1)
+
+
+def test_cleanup_after_failed_verification(mocker, testdir):
+    mocker.patch.object(Client, 'verify_email_code', None)
```

Comment:
This works, but I have a couple of comments. First, we're calling `Session.verify_email_code` but patching `Client.verify_email_code`. If the implementation of PyFxA changed, we might find this test is no longer doing what we intend. Ideally we'd mock the method that we're actually calling. Second, we're causing a `TypeError` exception because we're replacing the method with `None` instead of a mock object. This has the desired effect, but it would be better practive to create a mock object with a side effect of raising an exception. For example:

```python
 m = mocker.Mock(side_effect=Exception)
 mocker.patch.object(Session, 'verify_email_code', m)
```